### ISSUE TITLE
Incorrect "Kubernetes Descriptor" detection due to commented lines

### ISSUE THREAD

**Issue by ejether** [super-linter/super-linter#4350]
## Describe the bug

The "Kubernetes Descriptor" detection block [here](https://github.com/super-linter/super-linter/blob/main/lib/functions/detectFiles.sh#LL163C3-L169C5) detects commented lines as evidence that the file in question is a "Kubernetes Descriptor" . In our case that means that [this helm values file](https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/values.yaml#L742-L751) is detected as a "Kubernetes Descriptor" and fails because it is not.

I understand that  yaml manifest in a yaml values file is kind of a grey area, but I would expect that, while commented out, these lines wouldn't trigger kubeval,  or kubeconform, etc.

## Expected behavior

1. The "Kubernetes Descriptor"  detection logic should _not_ "hit" on a line that is commented out. 
2. Ideally it wouldn't "hit" on a helm values yaml file but that sounds tricky in this sort of case.

## Steps to Reproduce
1. `git clone git@github.com:jenkinsci/helm-charts.git`
2. `cd helm-charts`
3.  
```
docker run \
  -e RUN_LOCAL=true \
  -e ACTIONS_RUNNER_DEBUG=true \
  -e DISABLE_ERRORS=false \
  -e ERROR_ON_MISSING_EXEC_BIT=true \
  -e LINTER_RULES_PATH=. \
  -e MULTI_STATUS=false \
  -e VALIDATE_ALL_CODEBASE=true \
  -v $(pwd):/tmp/lint \
  ghcr.io/github/super-linter:v3.13.5
```
3. find `2023-06-01 20:17:03 [DEBUG]   /tmp/lint/charts/jenkins/values.yaml is a Kubernetes descriptor`
5. or errors like 
```
2023-06-01 20:17:23 [ERROR]   Error code: 1. Command run:[$ERR  - /tmp/lint/charts/jenkins/values.yaml: Missing 'metadata' key]
2023-06-01 20:17:23 [DEBUG]   Error code: 1. Command run:[$ERR  - /tmp/lint/charts/jenkins/values.yaml: Missing 'metadata' key].
```
Where the file is identified as a "Kubernetes Descriptor"

**Comment by ejether** [comment]
Fixed by https://github.com/super-linter/super-linter/pull/4375
